core/src/main/java/ai/timefold/solver/core/impl/move/director/VariableChangeRecordingScoreDirector.java

```diff
@@ -143,10 +173,23 @@ public SolutionDescriptor<Solution_> getSolutionDescriptor() {
         return delegate.getSolutionDescriptor();
     }
 
+    /**
+     * Returns the score director to which events are delegated.
+     */
     public InnerScoreDirector<Solution_, ?> getDelegate() {
         return delegate;
     }
 
+    /**
+     * The {@code VariableChangeRecordingScoreDirector} score director includes two main tasks:
+     * tracking any variable change and firing events to a delegated score director.
+     * This method returns a copy of the score director
+     * that only tracks variable changes without firing any delegated score director events.
+     */
+    public VariableChangeRecordingScoreDirector<Solution_> getNonDelegating() {
+        return new VariableChangeRecordingScoreDirector<>(delegate, variableChanges, cache, false);
```

Comment:
For safety, shouldn't `delegate` be `null` here instead?
That way, you can possibly also avoid the boolean - if the delegate is null, do not delegate.